voter-dapp/src/RetrieveRewards.js

```diff
@@ -43,24 +44,26 @@ function useRetrieveRewardsTxn(retrievedRewardsEvents, revealedVoteEvents, price
       voteState.retrievedRewards = true;
     }
 
-    for (const event of priceResolvedEvents) {
-      const voteState = getVoteState(event.returnValues.identifier, event.returnValues.time);
-
-      voteState.priceResolutionRound = event.returnValues.roundId.toString();
-    }
-
     // Since this loop is the last one, we can use the state to determine if this identifer, time, roundId combo
     // is eligible for a reward claim. We track the oldestUnclaimedRound to determine which round the transaction
     // should query (since only one can be chosen).
     let oldestUnclaimedRound = MAX_SAFE_JS_INT;
-    for (const event of revealedVoteEvents) {
+    for (let i = 0; i < revealedVoteEvents.length; i++) {
+      const event = revealedVoteEvents[i];
+      const priceRequest = priceRequests[i];
       const voteState = getVoteState(event.returnValues.identifier, event.returnValues.time);
       const revealRound = event.returnValues.roundId.toString();
+      const votedPrice = event.returnValues.price.toString();
 
-      if (!voteState.retrievedRewards && voteState.priceResolutionRound === revealRound) {
+      if (
+        !voteState.retrievedRewards &&
+        priceRequest.lastRound.toString() === revealRound &&
```

Comment:
Why do you have to make sure the last round was the Reveal round? Aren't rewards able to be retrieved once the `votedPrice === priceRequest.resolvedPrice.toString()` (i.e. once there exists a resolved price) ?

Reply:
This is to handle this case:
- You reveal the correct vote in round 5.
- The price request doesn't pass the GAT in round 5.
- You reveal the incorrect vote or don't reveal in round 6.
- The price request passes the GAT in round 6 resolving to your vote in round 5.

Put simply, the right vote at the wrong time means you don't get rewards :).

Reply:
Added a comment to this effect.

Reply:
So `lastRound` is when the price request was resolved, and `revealRound` is when the user revealed?

I think I got confused here because I though `revealRound` was opposite of `commitRound`, but I think looking at it now that `revealRound` is actually the round that a price was revealed.

Reply:
Yep, that's correct. Let me know if there's a comment I can add that would help make this clearer.